In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import warnings
warnings.filterwarnings('ignore')

# Import Dataset

In [3]:
df = pd.read_csv('hotel_bookings_data.csv')
df.head()

hotel  is_canceled  lead_time  arrival_date_year arrival_date_month  \
0  Resort Hotel            0        342               2017          September   
1  Resort Hotel            0        737               2017          September   
2  Resort Hotel            0          7               2017          September   
3  Resort Hotel            0         13               2017          September   
4  Resort Hotel            0         14               2017          September   

   arrival_date_week_number  arrival_date_day_of_month  \
0                        27                          1   
1                        27                          1   
2                        27                          1   
3                        27                          1   
4                        27                          1   

   stays_in_weekend_nights  stays_in_weekdays_nights  adults  ...  \
0                        0                         0       2  ...   
1                        0                         0       2  ...   
2                        0                         1       1  ...   
3                        0                         1       1  ...   
4                        0                         2       2  ...   

   booking_changes  deposit_type  agent company days_in_waiting_list  \
0                3    No Deposit    NaN     NaN                    0   
1                4    No Deposit    NaN     NaN                    0   
2                0    No Deposit    NaN     NaN                    0   
3                0    No Deposit  304.0     NaN                    0   
4                0    No Deposit  240.0     NaN                    0   

  customer_type   adr  required_car_parking_spaces  total_of_special_requests  \
0      Personal   0.0                            0                          0   
1      Personal   0.0                            0                          0   
2      Personal  75.0                            0                          0   
3      Personal  75.0                            0                          0   
4      Personal  98.0                            0                          1   

   reservation_status  
0           Check-Out  
1           Check-Out  
2           Check-Out  
3           Check-Out  
4           Check-Out  

[5 rows x 29 columns]

In [4]:
# Data Information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 29 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  object 
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  object 
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_weekdays_nights        119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal            

Dataset terdiri dari 29 kolom dan 119,390 baris.


# Mengatasi Missing Values

In [5]:
df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_weekdays_nights               0
adults                                 0
children                               4
babies                                 0
meal                                   0
city                                 488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
booking_changes                        0
deposit_type                           0
agent                              16340
company                           112593
days_in_waiting_list                   0
customer_type   

Drop kolom 'company' karena jumlah nilai null sangat banyak

In [6]:
df.drop(columns=['company'], inplace=True)

Drop nilai null pada kolom 'children' karena jumlahnya sedikit sehingga tidak mempengaruhi representatif dari data

In [7]:
df.dropna(subset=['children'], inplace=True)

Mengganti nilai null pada kolom 'city' dan 'agent' menjadi "Unknown"

In [8]:
df['city'].fillna(value='Unknown', inplace=True)
df['agent'].fillna(value='Unknown', inplace=True)

In [9]:
df.isna().sum()

hotel                             0
is_canceled                       0
lead_time                         0
arrival_date_year                 0
arrival_date_month                0
arrival_date_week_number          0
arrival_date_day_of_month         0
stays_in_weekend_nights           0
stays_in_weekdays_nights          0
adults                            0
children                          0
babies                            0
meal                              0
city                              0
market_segment                    0
distribution_channel              0
is_repeated_guest                 0
previous_cancellations            0
previous_bookings_not_canceled    0
booking_changes                   0
deposit_type                      0
agent                             0
days_in_waiting_list              0
customer_type                     0
adr                               0
required_car_parking_spaces       0
total_of_special_requests         0
reservation_status          

# Mengganti Nilai yang Tidak Sesuai

In [10]:
df['meal'].value_counts()

meal
Breakfast     92306
Dinner        14463
No Meal       10650
Undefined      1169
Full Board      798
Name: count, dtype: int64

Mengganti value 'Undefined' menjadi 'No Meal'

In [11]:
df['meal'].replace('Undefined', 'No Meal', inplace=True)

In [12]:
df['meal'].value_counts()

meal
Breakfast     92306
Dinner        14463
No Meal       11819
Full Board      798
Name: count, dtype: int64

# Membuang Data yang Tidak Perlu

Menjumlahkan kolom `adults, children, babies` menjadi `total_guests` kemudian menghilangkan kolom-kolom tersebut

In [25]:
df['total_guests'] = df['adults'] + df['children'] + df['babies']

In [28]:
df.drop(columns=['adults', 'children', 'babies'], inplace=True)